packages/sanity/src/core/store/_legacy/document/document-pair/validation.ts

```diff
@@ -114,81 +49,7 @@ export const validation = memoize(
       shareLatestWithRefCount(),
     )
 
-    const referenceIds$ = document$.pipe(
```

Comment:
all this logic below has been moved to [`validateDocumentWithReferences`](https://github.com/sanity-io/sanity/pull/7207/files#diff-f33b0c793316d3d8640c334231c4ecdef6eca59ba5b8b5bbab0698d46d33a906)